File: csharp/HearthstoneReplays/ReplayData/GameActions/HideEntity.cs
#region

using System.Xml.Serialization;

#endregion

namespace HearthstoneReplays.ReplayData.GameActions
{
	public class HideEntity : GameData
	{
		[XmlAttribute("entity")]
		public int Entity { get; set; }

		[XmlAttribute("tag")]
		public int TagName { get; set; }

		[XmlAttribute("value")]
		public int TagValue { get; set; }
	}
}

File: java/src/main/java/info/hearthsim/hsreplay/parser/replaydata/gameactions/HideEntity.java
package info.hearthsim.hsreplay.parser.replaydata.gameactions;

import info.hearthsim.hsreplay.parser.replaydata.GameData;

import javax.xml.bind.annotation.XmlAccessOrder;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorOrder;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@NoArgsConstructor
@XmlAccessorType(XmlAccessType.FIELD)
@XmlAccessorOrder(XmlAccessOrder.ALPHABETICAL)
@ToString
public class HideEntity extends GameData {

	@XmlAttribute(name = "entity")
	private int entity;

	@XmlAttribute(name = "zone")
	private int zone;

	public HideEntity(String timestamp, int entity, int zone) {
		super(timestamp);
		this.entity = entity;
		this.zone = zone;
	}

}
